api/v1alpha1/storageclient_types.go

```diff
@@ -46,10 +46,16 @@ type StorageClientSpec struct {
 	OnboardingTicket string `json:"onboardingTicket"`
 }
 
+type ProviderAttributes struct {
+	InMaintenanceMode bool `json:"inMaintenanceMode"`
+}
+
 // StorageClientStatus defines the observed state of StorageClient
 type StorageClientStatus struct {
 	Phase storageClientPhase `json:"phase,omitempty"`
 
+	Provider ProviderAttributes `json:"providerAttributes,omitempty"`
```

Comment:
1. I don't see why we need a sub struct here. 
2. I am not sure we want to use `provider` as terminology in our APIs

Reply:
Currently, we are sending one piece of information from the provider, in the future we might want multiple information. So, I think it would be good to have it under a substruct.